Gems/LyShine/Code/Editor/LyShineEditorSettings.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "LyShineEditorSettings.h"
+
+#include <AzCore/Settings/SettingsRegistry.h>
+
+namespace LyShine
+{
+
+    namespace Internal
+    {
+        template<typename T>
+        void SetRegistry(const AZStd::string_view setting, T&& value)
+        {
+            if (auto* registry = AZ::SettingsRegistry::Get())
+            {
+                registry->Set(setting, AZStd::forward<T>(value));
+            }
+        }
+
+        template<typename T>
+        AZStd::remove_cvref_t<T> GetRegistry(const AZStd::string_view setting, T&& defaultValue)
+        {
+            AZStd::remove_cvref_t<T> value = AZStd::forward<T>(defaultValue);
+            if (const auto* registry = AZ::SettingsRegistry::Get())
+            {
+                T potentialValue;
+                if (registry->Get(potentialValue, setting))
+                {
+                    value = AZStd::move(potentialValue);
+                }
+            }
+
+            return value;
+        }
+    }
+
+    ViewportInteraction::InteractionMode GetInteractionMode()
+    {
+        return static_cast<ViewportInteraction::InteractionMode>(
+            Internal::GetRegistry(InteractionModeSetting, aznumeric_cast<AZ::u64>(ViewportInteraction::InteractionMode::SELECTION)));
+    }
+
+    void SetInteractionMode(ViewportInteraction::InteractionMode mode)
+    {
+        Internal::SetRegistry(InteractionModeSetting, aznumeric_cast<AZ::u64>(mode));
+    }
+
+    ViewportInteraction::CoordinateSystem GetCoordinateSystem()
+    {
+        return static_cast<ViewportInteraction::CoordinateSystem>(
+            Internal::GetRegistry(CoordinateSystemSetting, aznumeric_cast<AZ::u64>(ViewportInteraction::CoordinateSystem::LOCAL)));
+    }
+
+    void SetCoordinateSystem(ViewportInteraction::CoordinateSystem system)
+    {
+        Internal::SetRegistry(CoordinateSystemSetting, aznumeric_cast<AZ::u64>(system));
+    }
```

Comment:
In theory you could write out the `/O3DE/Preferences/Editor/LyShine` section of the Settings Registry whenever SetInteractionMode or SetCoordinateMode is invoked automatically.

Or an UI option could be added to the EditorWindow to save the settings out.
The settings can be saved to any filename, as long as it is output to the `<project-path>/user/Registry` folder

Here is the documentation on dumping the SettingsRegistry to an `AZ::IO::GenericStream` derived class such as `AZ::IO::ByteContainerStream` or `AZ::IO::SystemFileStream`
https://www.o3de.org/docs/user-guide/settings/output-settings-registry/#storing-editor-preferences-using-a-key-and-an-anchor

```
AZ::SettingsRegistryMergeUtils::DumperSettings dumperSettings;
dumperSettings.m_prettifyOutput = true;
dumperSettings.m_jsonPointerPrefix = "/O3DE/Preferences/Editor/LyShine";

AZStd::string stringBuffer;
AZ::IO::ByteContainerStream stringStream(&stringBuffer);
if (auto registry = AZ::SettingsRegistry::Get();
    registry == nullptr || !AZ::SettingsRegistryMergeUtils::DumpSettingsRegistryToStream(*registry, "/O3DE/Preferences/Editor/LyShine", stringStream, dumperSettings))
{
    AZ_Warning("LyShine Editor Settings", false, R"(Unable to dump the LyShine Preferences settings from the Settings Registry)");
    return;
}

// stringBuffer now contains the dumped lyshine settings in JSON
// Now write it to disk

//! Get the path to the current projects `<project-path>/user` directory and append to the Registrt/lyshine_preferences.setreg file to it
auto lyShinePreferencesFilepath = AZ::IO::FixedMaxPath(AZ::Utils::GetProjectUserPath()) / "Registry/lyshine_preferences.setreg";

bool saved{};
constexpr auto openMode = AZ::IO::OpenMode::ModeWrite;
if (AZ::IO::SystemFileStream outputFileStream(lyShinePreferencesFilepath.c_str(), openMode);
    outputFileStream.IsOpen())
{
    saved = outputFile.Write(stringBuffer.size(), stringBuffer.data()) == stringBuffer.size();
}

AZ_Warning("LyShine Editor Settings", saved, R"(Unable to save LyShine Editor Preferences registry file to path "%s"\n)",
    lyShinePreferencesFilepath.c_str());
```

The Settings Registry will automatically load any .setreg/.setregpatch files from the current project's `<project-path>/user/Registry` directory that doesn't have Settings Registry specialization in it. 
NOTE: This only occurs automatically on application launch though.

Here is the [reference](https://www.o3de.org/docs/user-guide/settings/developer-documentation/#search-locations) to the search locations that applications look for .setreg/.setregpatch files and the order that they merge such files